### PR TITLE
adds natural sorting of report options

### PR DESCRIPTION
**Pull request by esotrope**
image

image

image

### CODE REVIEW

File: aggregate-service/src/main/resources/application.sql.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
 sql:
 
   administrationCondition:
-    findAll: select id, code from administration_condition
+    findAll: >-
+      SELECT id,
+        code
+      FROM administration_condition
+      ORDER BY id
```
Comment (esotrope) [inline aggregate-service/src/main/resources/application.sql.yml]: added explicit ordering of all entities returned as options for selection

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/DimensionType.java [hunk 2]
```diff
@@ -22,4 +37,12 @@ public static DimensionType caseInsensitiveValue(final String name) {
         }
         return null;
     }
+
+    public static List<DimensionType> orderedDisplayValues() {
+        return Arrays.stream(values())
+                .filter(type -> type != Overall)
+                .sorted((a, b) -> b.displayOrder - a.displayOrder)
+                .collect(toList());
+    }
```
Comment (esotrope) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/DimensionType.java]: went ahead and made the enum do all the work for preparing sorted display options.

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form-options.mapper.ts [hunk 3]
```diff
@@ -34,14 +34,12 @@ export class AggregateReportFormOptionsMapper {
           value: entity,
           text: this.translate.instant(`common.grade.${entity.code}.form-name`),
           analyticsProperties: { label: `Assessment Grade: ${entity.code}` }
-        })
-        .sort((a, b) => a.value.id - b.value.id),
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report-form-options.mapper.ts]: removed all client side sorting for options

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/model/DimensionType.java [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
+    Gender(1),
+    Ethnicity(2),
+    LEP(3),
+    MigrantStatus(4),
+    Section504(5),
+    IEP(6),
+    EconomicDisadvantage(7),
+    StudentEnrolledGrade(8);
+
+    private int displayOrder;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/model/DimensionType.java]: `final`

**Comment by wtritch** [comment]
I would still prefer that the default option for comparative subgroups be "None" and force the user to select all the ones they want (up to all of them)

**Comment by esotrope** [comment]
@wtritch i thought i commented on the last PR but maybe i didn't - the goal is to make assessment grades and comparative subgroups like that

**Pull request merged**